From d2d0a722df735de52015f0c8a01f27a9ec28622e Mon Sep 17 00:00:00 2001
From: Duc Dang <dhdang@apm.com>
Date: Tue, 1 Sep 2015 13:42:16 -0700
Subject: [PATCH 10/27] arm64: dts: Add poweroff device nodes for APM X-Gene 2
 platform

This patch adds:
 + poweroff button device node to support poweroff by
   power button on APM X-Gene 2 Merlin platform.
 + syscon poweroff device node to support poweroff feature
   on APM X-Gene 2 Merlin platform

Signed-off-by: Duc Dang <dhdang@apm.com>
---
 arch/arm64/boot/dts/apm/apm-merlin.dts | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/arch/arm64/boot/dts/apm/apm-merlin.dts b/arch/arm64/boot/dts/apm/apm-merlin.dts
index a20ef5b..119a469 100644
--- a/arch/arm64/boot/dts/apm/apm-merlin.dts
+++ b/arch/arm64/boot/dts/apm/apm-merlin.dts
@@ -23,6 +23,28 @@
 		device_type = "memory";
 		reg = < 0x1 0x00000000 0x0 0x80000000 >;
 	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+		button@1 {
+			label = "POWER";
+			linux,code = <116>;
+			linux,input-type = <0x1>;
+			interrupts = <0x0 0x28 0x1>;
+		};
+	};
+
+	poweroff_mbox: poweroff_mbox@10548000 {
+		compatible = "syscon";
+		reg = <0x0 0x10548000 0x0 0x30>;
+	};
+
+	poweroff: poweroff@10548010 {
+		compatible = "syscon-poweroff";
+		regmap = <&poweroff_mbox>;
+		offset = <0x10>;
+		mask = <0x1>;
+	};
 };
 
 &serial0 {
-- 
2.6.2

